gui/src/main/java/org/verapdf/gui/CheckerPanel.java

```diff
@@ -33,648 +35,654 @@
+					.endsWith(GUIConstants.DOT + extension.toLowerCase())) {
+				JOptionPane.showMessageDialog(
+						CheckerPanel.this,
+						"Error. Selected file is not in "
+								+ extension.toUpperCase() + " format.",
+						GUIConstants.ERROR, JOptionPane.ERROR_MESSAGE);
+			} else {
+
+				this.result = null;
+				this.resultLabel.setForeground(GUIConstants.BEFORE_VALIDATION_COLOR);
+				this.resultLabel.setText("");
+				this.xmlReport = null;
+				this.htmlReport = null;
+				this.saveXML.setEnabled(false);
+				this.viewXML.setEnabled(false);
+				this.saveHTML.setEnabled(false);
+				this.viewHTML.setEnabled(false);
+
+				switch (extension) {
+					case GUIConstants.PDF:
+						this.pdfFile = chooser.getSelectedFile();
+						this.chosenPDF.setText(this.pdfFile.getAbsolutePath());
+						break;
+					case GUIConstants.XML:
+						this.profile = chooser.getSelectedFile();
+						this.chosenProfile.setText(this.profile.getAbsolutePath());
+						break;
+					default:
+						// This method used only for previous two cases.
+						// So do nothing.
+				}
+				setValidationButtonEnability();
+			}
+		}
+	}
+
+	private void saveReport(JFileChooser chooser, String extension, File report) {
+		if (report == null) {
+			JOptionPane.showMessageDialog(CheckerPanel.this,
+					"Validation hasn't been run.", GUIConstants.ERROR,
+					JOptionPane.ERROR_MESSAGE);
+		} else {
+			chooser.setSelectedFile(new File(extension.toLowerCase()
+					+ "Report." + extension.toLowerCase()));
+			int resultChoose = chooser.showSaveDialog(CheckerPanel.this);
+			if (resultChoose == JFileChooser.APPROVE_OPTION) {
+				File temp = chooser.getSelectedFile();
+
+				if (!(temp.getName().toLowerCase().endsWith(GUIConstants.DOT
+						+ extension.toLowerCase()))) {
+					temp = new File(temp.getPath() + GUIConstants.DOT
+							+ extension.toLowerCase());
+				}
+
+				try {
+					try {
+						Files.copy(report.toPath(), temp.toPath());
+					} catch (FileAlreadyExistsException e1) {
+						LOGGER.debug(
+								"File already exists, conform overwrite with user",
+								e1);
+						int resultOption = JOptionPane
+								.showConfirmDialog(
+										CheckerPanel.this,
+										extension.toUpperCase()
+												+ " file with the same name already exists. Do you want to overwrite it?",
+										"", JOptionPane.YES_NO_OPTION);
+						if (resultOption == JOptionPane.YES_OPTION) {
+							Files.copy(report.toPath(), temp.toPath(),
+									StandardCopyOption.REPLACE_EXISTING);
+						}
+					}
+				} catch (IOException e) {
+					JOptionPane.showMessageDialog(CheckerPanel.this,
+							GUIConstants.ERROR_IN_SAVING_HTML_REPORT,
+							GUIConstants.ERROR, JOptionPane.ERROR_MESSAGE);
+					LOGGER.error("Exception saving " + extension.toUpperCase()
+							+ " report", e);
+				}
+			}
+		}
+	}
+
+	private void changeApplicationConfig() {
+		application.getConfig().setProcessingType((ProcessingType) processingType.getSelectedItem());
```

Comment:
if we are passing config to this object without cloning it, then we can just change values in this.config (because it is the same object)